internal/notifier/eventhub.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2020 The Flux authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"github.com/fluxcd/pkg/runtime/events"
+)
+
+type EventHub struct {
+	Hub *eventhub.Hub
+}
+
+func NewEventHub(endpointURL string) (*EventHub, error) {
+	hub, err := eventhub.NewHubFromConnectionString(endpointURL)
+	if err != nil {
+		return nil, err
+	}
+	return &EventHub{
+		Hub: hub,
+	}, nil
+}
+
+// Post EventHub msg
+func (e *EventHub) Post(event events.Event) error {
```

Comment:
I see that some of the notifiers are using pointers and some are not. Is there a reason for one or the other?

Reply:
What do you mean? I can't find any pointer with event events.Event. It wouldn't work.

Reply:
Above is using `func (e *Eventhub) ...`. 

The other notifiers are using a mix of `*` and not.

My question is (and this may be obvious and I just don't fully understand it) why are the different notifiers doing this differently and what should be used?

Reply:
We can take this offline